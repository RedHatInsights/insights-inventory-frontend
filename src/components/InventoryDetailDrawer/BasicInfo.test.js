import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { createPromise as promiseMiddleware } from 'redux-promise-middleware';
import BasicInfo from './BasicInfo';

describe('BasicInfo', () => {
  let initialState;
  let mockStore;
  beforeEach(() => {
    mockStore = configureStore([promiseMiddleware()]);
    initialState = {
      entityDetails: {
        entity: {
          display_name: 'something',
          id: 'some-id',
          tags: [
            { namespace: 'one', key: 'key', value: 'value' },
            { key: 'key', value: 'value' },
            { namespace: 'one', key: 'key' },
            { key: 'key' },
          ],
        },
      },
    };
  });

  it('should render without data', () => {
    const store = mockStore({
      entityDetails: {},
    });
    render(
      <Provider store={store}>
        <BasicInfo />
      </Provider>
    );

    expect(
      screen.queryByRole('heading', {
        name: /something/i,
      })
    ).not.toBeInTheDocument();
    expect(
      screen.getByRole('link', {
        name: /open in inventory/i,
      })
    ).toBeVisible();
  });

  it('should render with data', () => {
    const store = mockStore(initialState);
    render(
      <Provider store={store}>
        <BasicInfo />
      </Provider>
    );

    expect(
      screen.getByRole('heading', {
        name: /something/i,
      })
    ).toBeVisible();
    expect(
      screen.getByRole('link', {
        name: /open in inventory/i,
      })
    ).toBeVisible();
  });

  it('should render with no inv link', () => {
    const store = mockStore(initialState);
    render(
      <Provider store={store}>
        <BasicInfo hideInvLink />
      </Provider>
    );

    expect(
      screen.getByRole('heading', {
        name: /something/i,
      })
    ).toBeVisible();
    expect(
      screen.queryByRole('link', {
        name: /open in inventory/i,
      })
    ).not.toBeInTheDocument();
  });

  it('should render with tags', () => {
    const store = mockStore(initialState);
    const view = render(
      <Provider store={store}>
        <BasicInfo showTags />
      </Provider>
    );

    expect(view.asFragment()).toMatchSnapshot();
  });
});
