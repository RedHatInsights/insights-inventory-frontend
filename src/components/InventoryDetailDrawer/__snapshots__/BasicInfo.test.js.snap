// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfo should render with tags 1`] = `
<DocumentFragment>
  <div
    class="pf-l-stack pf-m-gutter"
  >
    <div
      class="pf-l-stack__item"
    >
      <div
        class="pf-l-split"
      >
        <div
          class="pf-l-split__item pf-m-fill"
        >
          <h3
            class="pf-c-title pf-m-xl"
            data-ouia-component-id="OUIA-Generated-Title-4"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            something
          </h3>
        </div>
        <div
          class="pf-l-split__item"
        >
          <a
            class="ins-c-entity-detail__inv-link"
            href="./insights/inventory/some-id"
          >
            Open in Inventory
          </a>
        </div>
      </div>
    </div>
    <div
      class="pf-l-stack__item"
    >
      <div
        class="pf-l-split pf-m-gutter"
      >
        <div
          class="pf-l-split__item"
        >
          Tags:
        </div>
        <div
          class="pf-l-split__item"
        >
          <div
            aria-label="Chip group category"
            class="pf-c-chip-group"
            data-ouia-component-type="PF4/ChipGroup"
            data-ouia-safe="true"
            role="group"
          >
            <div
              class="pf-c-chip-group__main"
            >
              <ul
                aria-label="Chip group category"
                class="pf-c-chip-group__list"
                role="list"
              >
                <li
                  class="pf-c-chip-group__list-item"
                >
                  <div
                    class="pf-c-chip"
                    data-ouia-component-id="OUIA-Generated-Chip-1"
                    data-ouia-component-type="PF4/Chip"
                    data-ouia-safe="true"
                  >
                    <span
                      class="pf-c-chip__text"
                      id="pf-random-id-1"
                    >
                      one/key=value
                    </span>
                  </div>
                </li>
                <li
                  class="pf-c-chip-group__list-item"
                >
                  <div
                    class="pf-c-chip"
                    data-ouia-component-id="OUIA-Generated-Chip-2"
                    data-ouia-component-type="PF4/Chip"
                    data-ouia-safe="true"
                  >
                    <span
                      class="pf-c-chip__text"
                      id="pf-random-id-2"
                    >
                      key=value
                    </span>
                  </div>
                </li>
                <li
                  class="pf-c-chip-group__list-item"
                >
                  <div
                    class="pf-c-chip"
                    data-ouia-component-id="OUIA-Generated-Chip-3"
                    data-ouia-component-type="PF4/Chip"
                    data-ouia-safe="true"
                  >
                    <span
                      class="pf-c-chip__text"
                      id="pf-random-id-3"
                    >
                      one/key
                    </span>
                  </div>
                </li>
                <li
                  class="pf-c-chip-group__list-item"
                >
                  <button
                    class="pf-c-chip pf-m-overflow"
                    data-ouia-component-id="OUIA-Generated-Chip-4"
                    data-ouia-component-type="PF4/OverflowChip"
                    data-ouia-safe="true"
                    type="button"
                  >
                    <span
                      class="pf-c-chip__text"
                    >
                      1 more
                    </span>
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
